tests/common/plugins/ptfadapter/__init__.py

```diff
@@ -3,12 +3,45 @@
 import pytest
 
 from ptfadapter import PtfTestAdapter
+import ptf.testutils
+
 
 DEFAULT_PTF_NN_PORT = 10900
 DEFAULT_DEVICE_NUM = 0
 ETH_PFX = 'eth'
 
 
+# Below code is to override the 'send' function in the ptf.testutils module. Purpose of this change is to insert
+# code for updating the packet pattern before send it out. Generally we want to make the payload part of injected
+# packet to have string of current test module and case name. While inspecting the captured packets, it is easier
+# to fiture out which packets are injected by which test case.
+def _send(test, port_id, pkt, count=1):
+    update_payload = getattr(test, "update_payload", None)
+    if update_payload and callable(update_payload):
+        pkt = test.update_payload(pkt)
+
+    return ptf.testutils.send_packet(test, port_id, pkt, count=count)
+setattr(ptf.testutils, "send", _send)
+
+
+# Below code is to override the 'dp_poll' function in the ptf.testutils module. This function is called by all
+# the other functions for receiving packets in the ptf.testutils module. Purpose of this overriding is to update
+# the payload of received packet using the same method to match the updated injected packets.
+def _dp_poll(test, device_number=0, port_number=None, timeout=-1, exp_pkt=None):
+    update_payload = getattr(test, "update_payload", None)
+    if update_payload and callable(update_payload):
+        exp_pkt = test.update_payload(exp_pkt)
+
+    result = test.dataplane.poll(
+        device_number=device_number, port_number=port_number,
+        timeout=timeout, exp_pkt=exp_pkt, filters=[]
```

Comment:
Should be filters=FILTERS?
```suggestion
        timeout=timeout, exp_pkt=exp_pkt, filters=[]
```

Reply:
Good catch! Updated.